### PR TITLE
Fix #264 - TimestampUtils uses non-threadsafe classes

### LINKED ISSUE #264: TimestampUtils uses non-threadsafe classes
**Summary:** 

This is possibly a cause behind some of the strange things we've seen lately related to #196 - "Elapsed time for each iteration doesn't seem right", and is likely the cause behind #168 "Incorrect flagging of invalid start_date format", at least the newest instance described in https://github.com/CUTR-at-USF/gtfs-realtime-validator/issues/168#issuecomment-313437398.

In `TimestampUtils`, we declare a few objects used for formatting dates and times:

~~~
private static DateFormat mDateFormat = new SimpleDateFormat("yyyyMMdd");
private static DateFormat mTimeFormat = new SimpleDateFormat("HH:mm:ss");
private static Pattern mTimePattern = Pattern.compile("^[0-2][0-9]:[0-5][0-9]:[0-5][0-9]$"); // Up to 29 hrs
private static DecimalFormat mDecimalFormat = new DecimalFormat("0.0##");
~~~

It looks like `Pattern` is thread-safe (https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html).

However, `DateFormat` and `DecimalFormat` are apparently not thread-safe, which could potentially cause corrupted output.

* For `DecimalFormat`, see http://jonamiller.com/2015/12/21/decimalformat-is-not-thread-safe/ - we could use `ThreadLocal` here.  I'd like to use the Java 8 convention for this as shown in https://stackoverflow.com/a/38069338/937715, like `private static ThreadLocal<NumberFormat> numberFormatter = ThreadLocal.withInitial(() -> new DecimalFormat("00000"));`
* For `SimpleDateFormat`, looks like `DateTimeFormatter` is the Java 8 thread-safe version - we should switch to using `DateTimeFormatter`.  https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

**Steps to reproduce:** 

Run several threads monitoring several feeds at once, and corrupted values may eventually be shown

It's very difficult to consistently reproduce, as it's caused by a multhreaded environment.

**Expected behavior:** 

Different threads should be able to format/check dates and times without potential for corruption.

**Observed behavior:** 

Multiple threads accessing `SimpleDateFormat` and `DecimalFormat ` corrupt the output

### PR DESCRIPTION
**Pull request by Suryakandukoori**
**Summary:**

Timestamp utils no longer user static Dateformat and Decimal format objects

**Expected behavior:** 

Multithreaded environment must not pose an issue to thread-safety of Dateformat and Decimal format objects

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Run the unit tests with `mvn test` to make sure you didn't break anything
- [X] Format the title like "Fix #issue - short description of fix and changes"
- [X] Linked all relevant issues
- [X] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: src/main/java/edu/usf/cutr/gtfsrtvalidator/background/BackgroundTask.java [hunk 1]
```diff
@@ -201,24 +202,25 @@ public void run() {
             // Use the same current time for all rules for consistency
             long currentTimeMillis = System.currentTimeMillis();
             consoleOutput = new StringBuffer();
+            timestampUtils = new TimestampUtils();
```
Comment (barbeau) [inline src/main/java/edu/usf/cutr/gtfsrtvalidator/background/BackgroundTask.java]: It seems wasteful to instantiate a new TimestampUtils class on each new thread execute.  I'd also prefer to keep utility classes as static that aren't instantiated.  Was there a reason why the alternate solutions outlined in https://github.com/CUTR-at-USF/gtfs-realtime-validator/issues/264 wouldn't work?

**Comment by Suryakandukoori** [comment]
@barbeau please review and merge the request

**Comment by barbeau** [comment]
Thanks, I'll take a look!

**Comment by barbeau** [comment]
Looks good, thanks!

**Pull request merged**